src/View/ViewBuilder.php

```diff
@@ -113,6 +113,77 @@ class ViewBuilder implements JsonSerializable, Serializable
      */
     protected $_helpers = [];
 
+    /**
+     * View vars
+     *
+     * @var array
+     */
+    protected $_vars = [];
+
+    /**
+     * Saves a variable for use inside a template.
+     *
+     * @param string $name A string or an array of data.
+     * @param mixed $value Value.
+     * @return $this
+     */
+    public function setVar($name, $value = null)
```

Comment:
Could the trait be updated to call this method if the implementing class has a `viewBuilder()` method? Would that make removing the public property easier later?

Reply:
Yes that's what I was thinking of doing, but then the viewVars property of the trait won't be populated. I personally have used the property to check if a bar is set so it would cause BC issue.

Reply:
Yup, so to avoid any BC issues let's keep ViewVarsTrait unchanged and simply mark the public $viewVars property as deprecated.

In 4.x we'll remove the property and update set() method to proxy the calls to view builder methods instead.

Reply:
Done